app/models/user.rb

```diff
@@ -12,4 +12,32 @@ class User < ActiveRecord::Base
   def current_commitment
     commitments.first
   end
+
+  def stats
```

Comment:
Non-blocking, but just something to think about.
There's enough logic here in the `#stats` method that this could be a separate object altogether. It wouldn't need to be an ActiveRecord-backed model, but it could be a Struct or a PORO that takes a User as a parameter and then collects all the stats into one place. This way, only the Stats object would need to change as features related to reporting stats are added (Single-Responsibility Principle).